--- configure.ac.old	2014-01-28 15:34:56.091187337 +0100
+++ configure.ac	2014-01-28 15:41:48.731185922 +0100
@@ -3,7 +3,7 @@
 m4_pattern_forbid([^_?DPKG_])
 
 AC_PREREQ(2.60)
-AC_INIT([dpkg], m4_esyscmd([./get-version]), [debian-dpkg@lists.debian.org])
+AC_INIT([dpkg], [1.17.6-reproducible-mempo0.1.23+1], [debian-dpkg@lists.debian.org])
 AC_CONFIG_SRCDIR([lib/dpkg/dpkg.h])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_AUX_DIR([build-aux])
